### ISSUE TITLE
target_include_directories do not use BUILD_INTERFACE

### ISSUE THREAD

**Issue by joejustesen** [nats-io/nats.c#489]
Instead of installing nats.c, I have fetched it using the CMake FetchContent as follows:

```
FetchContent_Declare(
        nats
        GIT_REPOSITORY https://github.com/nats-io/nats.c.git
        GIT_TAG        v3.2.0
)
FetchContent_MakeAvailable(nats)

```

The problem is where the nats target_include_directories is set for the shared and static library versions.   They are set as follows:

```
  target_include_directories(nats PUBLIC
        $<INSTALL_INTERFACE:include>)

  target_include_directories(nats_static PUBLIC
        $<INSTALL_INTERFACE:include>)
``` 
This does not seem to work with FetchContent as the INSTALL_INTERFACE is not where the code is located.  Adding the BUILD_INTERFACE to the target_include_directories seesm to fix this.  Like this:

```
 target_include_directories(nats PUBLIC
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
        $<INSTALL_INTERFACE:include>)

 target_include_directories(nats_static PUBLIC
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
        $<INSTALL_INTERFACE:include>)
```
Am I missing anything here?  If not, could this change be added?  Do you want a PR?

**Comment by kozlovic** [comment]
@joejustesen I really don't know. I have added that line in my local build and still am able to build on macOS, Linux and Windows VM, so I don't see any objection adding it. If you don't mind, please open a PR!

**Comment by joejustesen** [comment]
OK, I will have a PR in a day or so.

**Comment by joejustesen** [comment]
PR submitted.

**Comment by kozlovic** [comment]
Addressed with #496
